packages/waku/src/router/client.ts

```diff
@@ -318,9 +318,22 @@ const RouterSlot = ({
 const InnerRouter = ({ routerData }: { routerData: RouterData }) => {
   const refetch = useRefetch();
 
-  const [route, setRoute] = useState(() =>
-    parseRoute(new URL(window.location.href)),
+  const [route, setRoute] = useState(
+    () =>
+      ({
+        ...parseRoute(new URL(window.location.href)),
+        // Make sure there is no hash on the initial route,
+        // so it matches the server-rendered version. Otherwise
+        // there might be hydration errors.
+        hash: '',
+      }) as RouteProps,
   );
+
+  // Update the route post-load to include the current hash.
+  useEffect(() => {
+    setRoute(parseRoute(new URL(window.location.href)));
+  }, [setRoute]);
```

Comment:
I think we can omit `setRoute` in deps.
I wonder if we should add deps at all.

The safe bed would be to add `href`.
```suggestion
  }, [window.location.href]);
```